src/test/java/com/xceptance/xlt/engine/htmlunit/okhttp3/CookieJarImplTest.java

```diff
@@ -0,0 +1,155 @@
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie(domain, "name", "value", "/", null, false, false));
+
+        Assert.assertEquals("example.com", okCookie.domain());
+        Assert.assertEquals(expectedHostOnly, okCookie.hostOnly());
+    }
+
+    @Test
+    @Parameters(source = BlankStringParamProvider.class)
+    public void toOkHttpCookie_BlankName(final String name)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie("domain", name, "value", "/", null, false, false));
+
+        Assert.assertEquals("", okCookie.name());
+    }
+
+    @Test
+    @Parameters(source = NameProvider.class)
+    public void toOkHttpCookie_ValidName(final String name)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie("domain", name, "value", "/", null, false, false));
+
+        Assert.assertEquals("name", okCookie.name());
+    }
+
+    @Test
+    @Parameters(source = BlankStringOrNullParamProvider.class)
+    public void toOkHttpCookie_BlankValue(final String value)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie("domain", "name", value, "/", null, false, false));
+
+        Assert.assertEquals("", okCookie.value());
+    }
+
+    @Test
+    @Parameters(source = ValueProvider.class)
+    public void toOkHttpCookie_ValidValue(final String value)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie("domain", "name", value, "/", null, false, false));
+
+        Assert.assertEquals("value", okCookie.value());
+    }
+
+    @Test
+    @Parameters(source = BlankStringOrNullParamProvider.class)
+    public void toOkHttpCookie_BlankPath(final String path)
+    {
+        callToOkHttpCookieButExpectException(new Cookie("domain", "name", "value", path, null, false, false), CookieJarImpl.ERROR_MSG_PATH);
+    }
+
+    @Test
+    @Parameters(source = PathProvider.class)
+    public void toOkHttpCookie_ValidPath(final String path)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie("domain", "name", "value", path, null, false, false));
+
+        Assert.assertEquals("/path", okCookie.path());
+    }
+
+    @Test
+    public void toOkHttpCookie_InvalidPath()
+    {
+        callToOkHttpCookieButExpectException(new Cookie("domain", "name", "value", "path_not_starting_with_slash", null, false, false),
+                                             CookieJarImpl.ERROR_MSG_PATH);
+    }
+
+    private void callToOkHttpCookieButExpectException(final Cookie huCookie, final String expectedMessagePrefix)
+    {
+        try
+        {
+            CookieJarImpl.toOkHttpCookie(huCookie);
+            Assert.fail("Expected IllegalArgumentException");
+        }
+        catch (final IllegalArgumentException e)
+        {
+            Assert.assertTrue("Unexpected exception message", e.getMessage().startsWith(expectedMessagePrefix));
+        }
```

Comment:
This might be a cleaner solution for this:

```suggestion
        final IllegalArgumentException exception = Assert.assertThrows(IllegalArgumentException.class, () -> CookieJarImpl.toOkHttpCookie(huCookie));
        MatcherAssert.assertThat(exception.getMessage(), CoreMatchers.startsWith(expectedMessagePrefix));
```

(The MatcherAssert has a more helpful error message in case of failure, since it shows the actual and expected values)